compiler/src/dmd/dsymbolsem.d

```diff
@@ -7167,7 +7169,20 @@ private extern(C++) class SetFieldOffsetVisitor : Visitor
             assert(0, "unsupported bit-field style");
 
         const isMicrosoftStyle = style == TargetC.BitFieldStyle.MS;
-        const contributesToAggregateAlignment = target.c.contributesToAggregateAlignment(bfd);
+
+	/**
+	 * Indicates whether the specified bit-field contributes to the alignment
+	 * of the containing aggregate.
+	 * E.g., (not all) ARM ABIs do NOT ignore anonymous (incl. 0-length)
+	 * bit-fields.
+	 */
+        const bool contributesToAggregateAlignment = isMicrosoftStyle || !bfd.isAnonymous; // sufficient for supported architectures
```

Comment:
This _is_ target specific though. Each compiler back-end makes its own decision about this.

Reply:
For example, in GCC, back-ends set a `PCC_BITFIELD_TYPE_MATTERS` macro to notify us that type affects alignment. Then `TARGET_ALIGN_ANON_BITFIELD` if anonymous bitfields affect alignment as well.

Taking this away would likely regress data layouts on those back-ends.

Reply:
What matters is which ones are actually different.